scrapy/crawler.py

```diff
@@ -305,7 +305,7 @@ def start(self, stop_after_crawl=True):
             d.addBoth(self._stop_reactor)
 
         resolver_class = load_object(self.settings["DNS_RESOLVER"])
-        resolver_class.install(reactor, self.settings)
+        resolver_class.install_on_reactor(reactor, crawler=self)
```

Comment:
I think calling create_instance here would be a bit more clear: this would match how create_instance is used on other places, and allow to remove a duplicate `resolver = create_instance(cls, None, crawler, reactor=reactor)` line which every resolved implements.

Reply:
The two resolvers are installed using different reactor methods (`installResolver` and `installNameResolver`). With this approach, we are taking the responsibility of knowing the appropriate installer to use to the resolver itself (in case someone wants to implement their own resolver, for instance).
What we can do is transform this into an instance method (perhaps that's what you meant :sweat_smile: )

Reply:
Something like https://github.com/scrapy/scrapy/pull/4227/commits/3cfa73b8b12dfe2dd1364af856d2ec46a486b48d